packages/flutter/test/widgets/widget_inspector_structure_error_test.dart

```diff
@@ -2,93 +2,85 @@
+          context: ErrorDescription('during layout'),
+          exception: StackTrace.current,
+        );
+        FlutterError.reportError(expectedError);
 
+        // Verify structured errors are not shown.
+        expect(onErrorCalled, true);
+        expect(service.getEventsDispatched('Flutter.Error'), isEmpty);
 
-  static void runTests() {
-    final StructureErrorTestWidgetInspectorService service = StructureErrorTestWidgetInspectorService();
-    WidgetInspectorService.instance = service;
+        // Set callback that calls presentError.
+        onErrorCalled = false;
+        FlutterError.onError = (FlutterErrorDetails details) {
+          FlutterError.presentError(details);
+          onErrorCalled = true;
+        };
 
-    test('ext.flutter.inspector.structuredErrors reports error to _structuredExceptionHandler on error', () async {
-      final FlutterExceptionHandler? oldHandler = FlutterError.onError;
+        FlutterError.reportError(expectedError);
 
-      bool usingNewHandler = false;
-      // Creates a spy onError. This spy needs to be set before widgets binding
-      // initializes.
-      FlutterError.onError = (FlutterErrorDetails details) {
-        usingNewHandler = true;
-      };
+        // Verify structured errors are shown.
+        expect(onErrorCalled, true);
+        // Structured errors are not supported on web.
+        if (!kIsWeb) {
+          expect(service.getEventsDispatched('Flutter.Error'), hasLength(1));
+        } else {
+          expect(service.getEventsDispatched('Flutter.Error'), isEmpty);
+        }
 
-      WidgetsFlutterBinding.ensureInitialized();
-      try {
-        // Enables structured errors.
+        // Verify disabling structured errors sets the default FlutterError.presentError
         expect(
           await service.testBoolExtension('structuredErrors', <String, String>{'enabled': 'true'}),
           equals('true'),
         );
+        expect(FlutterError.presentError, isNot(equals(oldHandler)));
 
-        // Creates an error.
-        final FlutterErrorDetails expectedError = FlutterErrorDetails(
-          library: 'rendering library',
-          context: ErrorDescription('during layout'),
-          exception: StackTrace.current,
+        expect(
+          await service.testBoolExtension('structuredErrors', <String, String>{'enabled': 'false'}),
+          equals('false'),
         );
-        FlutterError.reportError(expectedError);
-
-        // For non-web apps, this validates the new handler did not receive an
-        // error because `FlutterError.onError` was set to
-        // `WidgetInspectorService._structuredExceptionHandler` when service
-        // extensions were initialized. For web apps, the new handler should
-        // have received an error because structured errors are disabled by
-        // default on the web.
-        expect(usingNewHandler, equals(kIsWeb));
+        expect(FlutterError.presentError, equals(oldHandler));
```

Comment:
I just noticed that this part of the test doesn't fail on web. Should `isStructuredErrorsEnabled` be used to prevent this?

Reply:
/cc @kenzieschmoll Is the comment about this not working on the web still accurate?

Reply:
It will prevent this but not with the test binding you are using which always returns true.
Here is the relevant line that disables it on the web.
https://github.com/flutter/flutter/blob/975fcc1f2fc6f4dc237363cd0b82e0a3ea6cc14f/packages/flutter/lib/src/widgets/widget_inspector.dart#L970
you could duplicate that logic in the test binding used by this case to reproduce it.

Reply:
However [this line](https://github.com/flutter/flutter/blob/bd4a376e009dfaf934648c66b19d836fd15dedf3/packages/flutter/test/widgets/widget_inspector_structure_error_test.dart#L67) from above fails on when run on web. Sorry I probably just wasn't clear, what I meant is should the `isStructuredErrorsEnabled` be used [here](https://github.com/flutter/flutter/blob/bd4a376e009dfaf934648c66b19d836fd15dedf3/packages/flutter/lib/src/widgets/widget_inspector.dart#L1000) as well as [here](https://github.com/flutter/flutter/blob/bd4a376e009dfaf934648c66b19d836fd15dedf3/packages/flutter/lib/src/widgets/widget_inspector.dart#L984)?